bundles/org.openhab.binding.pegelonline/src/main/java/org/openhab/binding/pegelonline/internal/handler/PegelOnlineHandler.java

```diff
@@ -0,0 +1,168 @@
+import org.openhab.binding.pegelonline.internal.dto.Measure;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.unit.MetricPrefix;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PegelOnlineHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class PegelOnlineHandler extends BaseThingHandler {
+    private static final String STATIONS_URI = "https://www.pegelonline.wsv.de/webservices/rest-api/v2/stations";
+    private final Logger logger = LoggerFactory.getLogger(PegelOnlineHandler.class);
+    private Optional<PegelOnlineConfiguration> configuration = Optional.empty();
+    private Optional<ScheduledFuture<?>> schedule = Optional.empty();
+    private Optional<Measure> cache = Optional.empty();
+    private TreeMap<Integer, Integer> warnMap = new TreeMap<>();
+    private String stationUUID = UNKNOWN;
+    private HttpClient httpClient;
+
+    public PegelOnlineHandler(Thing thing, HttpClient hc) {
+        super(thing);
+        httpClient = hc;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            if (cache.isPresent()) {
+                Measure m = cache.get();
+                if (LEVEL_CHANNEL.equals(channelUID.getId())) {
+                    updateChannelState(LEVEL_CHANNEL, QuantityType.valueOf(m.value, MetricPrefix.CENTI(SIUnits.METRE)));
+                } else if (TREND_CHANNEL.equals(channelUID.getId())) {
+                    updateChannelState(TREND_CHANNEL, DecimalType.valueOf(Integer.toString(m.trend)));
+                } else if (TIMESTAMP_CHANNEL.equals(channelUID.getId())) {
+                    updateChannelState(TIMESTAMP_CHANNEL, DateTimeType.valueOf(m.timestamp));
+                } else if (WARNING_CHANNEL.equals(channelUID.getId())) {
+                    updateChannelState(WARNING_CHANNEL,
+                            DecimalType.valueOf(Integer.toString(warnMap.floorEntry((int) m.value).getValue())));
+                }
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        PegelOnlineConfiguration config = getConfigAs(PegelOnlineConfiguration.class);
+        configuration = Optional.of(config);
```

Comment:
It would be nice if some config validation is added and if it fails the thing can be set to configuration error.

Reply:
validation of required parameter `uuid` added

Reply:
- the current implementation only checks the uid, not the other parameters for sanity.
- usually i prefer a check within the config class as that class is responsible for the configuration. In this case a remote api call is performed so it is ok to have it elsewhere.
- the api call is blocking and that is not allowed. the init function should return fast.

I would propose to have a validation method in the config class that checks the values for sanity (without remote calls), like having a value, check fo length, format, range etc. Resulting in a initilize method that does not check if the station uuid is actually valid. 
When the `performMeasurement` method is called, i expect some kind of error message if the uuid is not valid. That method could throw an exception or set the ThingStatusDetail to configuration exception

Reply:
> * the current implementation only checks the uid, not the other parameters for sanity.

Hard to check the other parameters for sanity if  there's no indication of a correct value. I put in a check for warnings to validate higher warnings don't have lower measure levels.

> * usually i prefer a check within the config class as that class is responsible for the configuration

Done



> * The init function should return fast.

Done


> checks the values for sanity (without remote calls), like `string.isBlank()`,

`uuid` is checked now for lower case alphanumeric & hyphen - this fits to all available stations and is performed as *quick check*.
If `performMeasurement` answers with 404 the uuid isn't correct and `CONFIGURATION_ERROR` status will be shown